locations/spiders/bananarepublic.py

```diff
@@ -1,120 +1,25 @@
-                this_day_group = {"from_day": day, "to_day": day, "hours": hours}
-            elif this_day_group["hours"] == hours:
-                this_day_group["to_day"] = day
-
-        if this_day_group:
-            day_groups.append(this_day_group)
-
-        opening_hours = ""
-        if len(day_groups) == 1 and day_groups[0]["hours"] in (
-            "00:00-23:59",
-            "00:00-00:00",
-        ):
-            opening_hours = "24/7"
-        else:
-            for day_group in day_groups:
-                if day_group["from_day"] == day_group["to_day"]:
-                    opening_hours += "{from_day} {hours}; ".format(**day_group)
-                elif day_group["from_day"] == "Mo" and day_group["to_day"] == "Su":
-                    opening_hours += "{hours}; ".format(**day_group)
-                else:
-                    opening_hours += "{from_day}-{to_day} {hours}; ".format(**day_group)
-            opening_hours = opening_hours[:-2]
-        return opening_hours
+    item_attributes = {"brand": "Banana Republic", "brand_wikidata": "Q806085"}
+    sitemap_urls = [
+        "https://bananarepublic.gap.com/stores/sitemap.xml",
+        "https://bananarepublic.gapcanada.ca/stores/sitemap.xml",
+    ]
+    sitemap_rules = [(".html", "parse"), ("ca/stores", "parse")]
+    download_delay = 0.5
 
     def parse(self, response):
-        data = response.xpath('//div[@id="sdStoreStates"]//li/a/@href').extract()
-        for store in data:
-            match = re.search(r"^.+(-store)\-(\d{1,4})(.jsp)$", store)
-            (_, store_id, _) = match.groups()
-            yield scrapy.Request(
-                self.store_url.format(store_id), callback=self.parse_store
-            )
-
-    def parse_store(self, response):
-        store = response.json()["storeLocations"]["storeLocationList"]
-        store_addr = store["storeAddress"]
-        addr1 = store_addr["addressLine1"]
-        (num, street) = (addr1, addr1)
-        if " " in addr1:
-            (num, street) = addr1.split(" ", 1)
-        zip_code = store_addr["postalCode"]
-        properties = {
-            "phone": store_addr["phoneNumber"],
-            "ref": store["storeId"],
-            "name": store["storeName"],
-            "opening_hours": self.store_hours(store.get("storeHours", None)),
-            "lat": store["latitude"],
-            "lon": store["longitude"],
-            "addr_full": store_addr["addressLine1"],
-            "housenumber": num,
-            "street": street,
-            "city": store_addr["cityName"],
-            "state": store_addr["stateProvinceCode"],
-            "postcode": zip_code,
-            "country": store_addr["countryCode"],
-            "website": response.url,
-        }
-
-        yield GeojsonPointItem(**properties)
+        if item := LinkedDataParser.parse(response, "ClothingStore"):
+            if item.get("lat"):
+                item["name"] = "Banana Republic"
+                item["ref"] = response.url
```

Comment:
I would prefer we just use the unique store id from the URL as the ref.

Reply:
Really .... do you think it is unique across their estate? That is kind of asking for trouble. "ref" I think is primarily for duplicate removal and if they do not supply then the clue is in the name of URL. I think LinkedDataParser and friends if they cannot get a "ref" supplied by the website should use the response.url ... or at least the path part.

Reply:
> do you think it is unique across their estate?

Yes. And I think when we're writing a spider we should look for a uid. Ideally that would be published in the data, but if not, we should find one.

> That is kind of asking for trouble. [snip] cannot get a "ref" supplied by the website should use the response.url 

We've already ran into trouble using the whole URL as the ref. #3788

> or at least the path part

That's what I'm asking for, the uid is in the path.

It's really not that uncommon for these websites to ignore everything except the uid too. e.g. Holland and Barratt [NL store](https://www.hollandandbarrett.nl/winkels/genk-9033/), take the uid (`genk-9033`) and add checkout the [GB website](https://www.hollandandbarrett.com/stores/genk-9033/). Using the whole url (or even the whole path) as a ref, this gives us 2 stores, not de duped. Use the uid and one gets removed as a duplicate.

Reply:
There is no perfect solution.

In this case we use direct store sitemap URL ... which is preferred. So having to even fart around with a "ref" at all is somewhat obtuse.

The code as it stands works.

Reply:
In this case, we can't copy the uid from one site to another. It's not the end of the world to use the whole url as the ref.